index.js

```diff
@@ -358,26 +358,28 @@ const createFocusTrap = function (elements, userOptions) {
          * @returns {HTMLElement|undefined} The next tabbable node, if any.
          */
         nextTabbableNode(node, forward = true) {
-          const nodeIdx = tabbableNodes.findIndex((n) => n === node);
+          const nodeIdx = tabbableNodes.indexOf(node);
           if (nodeIdx < 0) {
             // either not tabbable nor focusable, or was focused but not tabbable (negative tabindex):
             //  since `node` should at least have been focusable, we assume that's the case and mimic
             //  what browsers do, which is set focus to the next node in __document position order__,
             //  regardless of positive tabindexes, if any -- and for reasons explained in the NOTE
             //  above related to `firstDomTabbable` and `lastDomTabbable` properties, we fall back to
             //  basic DOM order
-            return forward ? firstDomTabbableNode : lastDomTabbableNode;
-          }
+            const nodeFocusableIndex = focusableNodes.indexOf(node);
+
+            if (forward) {
+              return focusableNodes
+                .slice(nodeFocusableIndex + 1)
+                .find((el) => isTabbable(el));
+            }
 
-          if (forward) {
-            // first found, if any (array will be empty if `node` was in last position)
-            return tabbableNodes.slice(nodeIdx + 1).find(() => true);
+            return focusableNodes
+              .slice(0, nodeFocusableIndex)
+              .findLast((el) => isTabbable(el));
           }
 
-          return tabbableNodes
-            .slice(0, nodeIdx) // exclude `node` itself
-            .reverse()
-            .find(() => true); // first found, if any (array will be empty if `node` was in first position)
+          return tabbableNodes[nodeIdx + (forward ? 1 : -1)];
```

Comment:
This could lead to a positive or negative index that doesn't exist in the `tabbableNodes` array. Technically, the result should be `undefined`, but it feels a bit wrong to access an array out of bounds.

Reply:
As a person who learned to code with C, I feel you. But in JavaScript it's a safe operation.
In any case I pushed f198c2b right while you writing this comment :smile: and the out of bound access is gone already.

Reply:
I started with C++. 😉

Reply:
After reverting the changes: c81b506

 We got the out of bound access again: `return tabbableNodes[nodeIdx + (forward ? 1 : -1)];`
Even though it's not undefined behaviour in JS, I understand that it's not good looking, but neither is the code it is simplifying:

```js
if (forward) {
  // first found, if any (array will be empty if `node` was in last position)
  return tabbableNodes.slice(nodeIdx + 1).find(() => true);
}

return tabbableNodes
  .slice(0, nodeIdx) // exclude `node` itself
  .reverse()
  .find(() => true); // first found, if any (array will be empty if `node` was in first position)

```

At the end it's just a question of preference, the library would work either way, so if you prefer that way I revert the changes.